### ISSUE TITLE
Interceptor response headers are dropped

### ISSUE THREAD

**Issue by ncskier** [tektoncd/triggers#392]
# Expected Behavior
I expect the Headers to persist across a chain of Interceptors without each Interceptor having to write every header back to the response.

Our [interceptor documentation](https://github.com/tektoncd/triggers/blob/master/docs/eventlisteners.md#event-interceptor-services) does not say that an interceptor must write all of the Headers from the request back to the response:
> To be an Event Interceptor, a Kubernetes object should:
> - Be fronted by a regular Kubernetes v1 Service over port 80
> - Accept JSON payloads over HTTP
> - Return a HTTP 200 OK Status if the EventListener should continue processing the event
> - Return a JSON body back. This will be used by the EventListener as the event payload for any further processing. If the interceptor does not need to modify the body, it can simply return the body that it received.

# Actual Behavior
In a chain of Interceptors, the headers are lost if one Interceptor does not write the headers to its http response. For example, with a chain of 3 Interceptors with the following code, only the first Interceptor in the chain receives the proper request headers:
```golang
func main() {
	http.HandleFunc("/", func(writer http.ResponseWriter, request *http.Request) {
		// Read and print the header
		log.Print("")
		log.Printf("Header: %v", request.Header)

		// Read and print the request body
		body, err := ioutil.ReadAll(request.Body)
		if err != nil {
			log.Printf("Error reading body: %s", err)
			http.Error(writer, "cannot read body", http.StatusBadRequest)
		}
		log.Printf("Body: %s", string(body))

		// Create response
		if _, err := writer.Write(body); err != nil {
			log.Printf("Error writing response: %s", err)
		}
	})

	log.Print("Listen and serve on port 8080")
	log.Fatal(http.ListenAndServe(":8080", nil))
}
```
Here are the logs from the interceptor when header `"foo": "bar"` was included in the request:
```
Header: map[Accept:[*/*] Accept-Encoding:[gzip] Content-Type:[application/json] Foo:[bar] Header-K0:[test-interceptors-0] User-Agent:[curl/7.64.1]]
...
Header: map[Accept-Encoding:[gzip] Content-Type:[text/plain; charset=utf-8] Date:[Tue, 28 Jan 2020 Header-K0:[test-interceptors-1] User-Agent:[Go-http-client/1.1]]
...
Header: map[Accept-Encoding:[gzip] Content-Type:[text/plain; charset=utf-8] Date:[Tue, 28 Jan 2020 21:42:52 GMT] Header-K0:[test-interceptors-2] User-Agent:[Go-http-client/1.1]]
```
The interceptor only received the `"foo": "bar"` header the first time it was fired. The header information was dropped on the second and third times.

# Steps to Reproduce the Problem

1. Create an EventListener with a chain of interceptors
2. Send a request with a header to the EventListener

# Additional Info
I'm not sure if this is necessarily a bug; we might just need to update the interceptor documentation. 
However, it doesn't look like the [GitHub interceptor](https://github.com/tektoncd/triggers/blob/master/pkg/interceptors/github/github.go#L49) writes its request headers to the response. So, it might be good to check that the built-in interceptors don't encounter this issue when they're included in an interceptor chain.

**Comment by dibyom** [comment]
I think the behavior is that we pass along each request in its entirety (body and headers) to the next interceptor in the chain. The built in interceptors should all return back the headers that were passed in e.g. [Github](https://github.com/tektoncd/triggers/blob/master/pkg/interceptors/github/github.go#L92) If they don't that's definitely a bug.

Where things get a bit weird are Webhook interceptors (esp if you add in the headers that get added to the requests sent to the interceptor). We should definitely update the docs and release notes to state that Webhook interceptors need to return the headers (this is a breaking change in the API).

**Comment by dibyom** [comment]
related #271

**Comment by dibyom** [comment]
This should be fixed now with the ongoing work on #271
